src/main/java/algorithms/curated170/easy/Permutations.java

```diff
@@ -0,0 +1,34 @@
+package algorithms.curated170.easy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Permutations {
+        public List<List<Integer>> permute(int[] nums) {
+            List<List<Integer>> output = new ArrayList<List<Integer>>();
```

Comment:
Passing output as a parameter instead of using a global variable looks more appropriate for "the Law of Demeter"

The class itself does not need to know about output at all.

Global variables are also dangerous in more complex projects as the state of a global variable can be changed by an unrelated method or class accidentally.

In short, it is easier to isolate a variable and track its state using it as a parameter.

Similar arguments against global variables can be found:

https://stackoverflow.com/questions/2216340/the-advantage-disadvantage-between-global-variables-and-function-parameters-in

Reply:
They do make the code simpler in small systems like this though. More "functional" approaches might be more suitable for the reliability of complex systems but I think such object oriented approaches okay for here.